### PR TITLE
Fetch vat from NstJoin

### PR DESCRIPTION
**Pull request by oldchili**

### CODE REVIEW

**Review by telome — APPROVED** [review]
LGTM

File: src/AllocatorVault.sol [hunk 1]
```diff
@@ -81,19 +81,17 @@ contract AllocatorVault {
...
         roles = RolesLike(roles_);
 
         buffer = buffer_;
-        vat = VatLike(vat_);
         ilk = ilk_;
         nstJoin = NstJoinLike(nstJoin_);
 
-        require(vat_ == nstJoin.vat(), "AllocatorVault/vat-not-match");
-
+        vat = VatLike(nstJoin.vat());
```
Comment (sunbreak1211) [inline src/AllocatorVault.sol]: I'd say to keep more symmetry with the `nst()` call, let's make `nstJoin.vat()` to return directly `VatLike` type.

Comment (oldchili) [reply src/AllocatorVault.sol]: done

**Review by sunbreak1211 — APPROVED** [review]
LGTM

**Pull request merged**